csharp/src/Ice/Communicator-Properties.cs

```diff
@@ -144,6 +144,58 @@ internal PropertyValue(string v, bool u)
             }
         }
 
+        /// <summary>Gets the value of a property as a TimeSpan. If the property is not set, returns null.
+        /// The value must be an integer followed immediately by a time unit of 'ms', 's', 'm', or 'h'.
+        /// These correspond to milliseconds, seconds, minutes, and hours, respectively.
+        /// e.g. 50ms, 3m</summary>
+        /// <param name="name">The property name.</param>
+        /// <returns>The property value parsed into a TimeSpan or null.</returns>
+        public TimeSpan? GetPropertyAsTimeSpan(string name)
+        {
+            lock (_properties)
+            {
+                if (_properties.TryGetValue(name, out PropertyValue? pv))
+                {
+                    pv.Used = true;
+
+                    if (pv.Val == "infinite") {
+                        return TimeSpan.FromMilliseconds(-1);
+                    }
+
+                    // Match an integer followed letters
+                    Match match = Regex.Match(pv.Val, @"^(\d+)([a-z]+)$");
+
+                    if (!match.Success)
+                    {
+                        throw new InvalidConfigurationException(
+                            $"the value of property `{name}' is not a TimeSpan");
+                    }
+
+                    int value;
+                    try
+                    {
+                        value = int.Parse(match.Groups[1].Value, CultureInfo.InvariantCulture);
+                    }
+                    catch (FormatException ex)
+                    {
+                        throw new InvalidConfigurationException(
+                            $"the value of property `{name}' is not a TimeSpan", ex);
+                    }
+
+                    return match.Groups[2].Value switch
```

Comment:
I feel this can go in the above try, and then you can declare value when we assign it.